### PR TITLE
test(exec_command): add output collection regression tests

### PR DESCRIPTION
**Pull request by clouatre**
## Summary

Adds four integration tests to `crates/aptu-coder/tests/exec_command.rs` that guard the output collection path in `run_exec_impl` against future regressions like the one fixed in #806.

## Motivation

The `rx.close()` regression (#804 → fixed in #806) produced no test failure because no existing test asserted on stdout/stderr being non-empty after the full MCP round-trip. These tests close that gap.

## Tests added

All four use the existing `call_exec_command_raw` helper (full MCP protocol: initialize → tools/call → structuredContent assertions).

| Test | Behavior | Type |
|---|---|---|
| `test_output_collection_stdout_nonempty` | `echo` produces non-empty stdout | happy path |
| `test_output_collection_stderr_nonempty` | stderr-only command populates stderr, not stdout | happy path |
| `test_output_collection_empty_output` | `true` exits 0, no overflow paths set | edge case |
| `test_output_collection_partial_before_timeout` | output printed before a 2-second timeout is captured | edge case |

## Changes

- `crates/aptu-coder/tests/exec_command.rs`: +84 lines (4 tests appended)

## Test plan

- [x] New tests pass (`cargo test --package aptu-coder --test exec_command test_output_collection`)
- [x] Full suite passes (472 passed, 0 failed)
- [x] Linter clean (`cargo clippy -- -D warnings`)
- [x] Formatter clean (`cargo fmt --check`)

## Related

Depends on #806 (fix). Should be merged after #806.

### CODE REVIEW

File: crates/aptu-coder/tests/exec_command.rs [hunk 1]
```diff
@@ -855,3 +855,91 @@ async fn test_exec_slot_files_not_written_for_small_output() {
...
+// --- Output collection regression tests (PR #804 / fix #806) ---
+// These four tests cover the behaviors that were silently broken by the
+// premature rx.close() regression: non-empty stdout, non-empty stderr,
+// empty output (no output at all), and partial output captured before timeout.
+
+#[tokio::test]
+async fn test_output_collection_stdout_nonempty() {
+    // Arrange: command that prints a known string to stdout
+    let resp =
+        call_exec_command_raw(serde_json::json!({"command": "echo regression_marker"})).await;
```
[the remark this answers is not in the document]
Comment (clouatre) [reply crates/aptu-coder/tests/exec_command.rs:875]: Added `sc_str(sc, key)` helper above `call_exec_command_raw`. All four new tests now use it, eliminating the `.as_str().unwrap_or("")` repetition.

**Comment by clouatre** [comment]
Closing. The tests do not catch the rx.close() hazard they were written for (verified by re-inserting the regression -- all 4 tests pass regardless). Coverage of exec_command output collection is already provided by the existing test suite. No net value over what is already present.
